rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -104,9 +104,13 @@ class NodeBase : public NodeBaseInterface
   get_associated_with_executor_atomic() override;
 
   RCLCPP_PUBLIC
-  rcl_guard_condition_t *
+  const rcl_guard_condition_t *
   get_notify_guard_condition() override;
 
+  RCLCPP_PUBLIC
+  rclcpp::GuardCondition *
+  get_guard_condition() override;
+
```

Comment:
I created the new `get_guard_condition()` because in the original PR william said:

> Also, anywhere the API is changed in a way that would break existing code calling it in one of our .hpp files, we need to consider deprecating it rather than just changing it. It isn't always possible, desirable, or necessary, but we need to think about each case and justify it if we don't deprecate things.

I think he was refering to I changed the original `get_notify_guard_condition` to return a `rclcpp::GuardCondition *`, but I'm not sure